server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -140,6 +162,12 @@ private void killUnusedSegments(Collection<String> dataSourcesToKill)
           break;
         }
       }
+
+      if (submittedTasks >= availableKillTaskSlots) {
+        log.info("Reached kill task slot limit with pending unused segments to kill. Will resume "
+                 + "on the next coordinator cycle.");
+        break;
+      }
     }
 
     log.debug("Submitted kill tasks for [%d] datasources.", submittedTasks);
```

Comment:
Out of scope: is this debug statement accurate? I believe it should be:
```suggestion
    log.debug("Submitted [%d] kill tasks for [%d] datasources.", submittedTasks, dataSourcesToKill.size());
```

Reply:
fixed